src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -80,6 +81,17 @@
         return currentBiomes;
     }
 
+    private static List<OceanBiomeEntry>[] setupOceanBiomes()
+    {
+        List<OceanBiomeEntry>[] currentBiomes = new List[OceanType.values().length];
+        currentBiomes[OceanType.WARM.ordinal()] = Lists.newArrayList(new OceanBiomeEntry(Biomes.WARM_OCEAN, Biomes.DEEP_WARM_OCEAN, Biomes.LUKEWARM_OCEAN, 10));
+        currentBiomes[OceanType.LUKEWARM.ordinal()] = Lists.newArrayList(new OceanBiomeEntry(Biomes.LUKEWARM_OCEAN, Biomes.DEEP_LUKEWARM_OCEAN, Biomes.DEEP_LUKEWARM_OCEAN, 10));
+        currentBiomes[OceanType.FROZEN.ordinal()] = Lists.newArrayList(new OceanBiomeEntry(Biomes.FROZEN_OCEAN, Biomes.DEEP_FROZEN_OCEAN, Biomes.COLD_OCEAN, 10));
+        currentBiomes[OceanType.COLD.ordinal()] = Lists.newArrayList(new OceanBiomeEntry(Biomes.COLD_OCEAN, Biomes.DEEP_COLD_OCEAN, Biomes.DEEP_COLD_OCEAN, 10));
+        currentBiomes[OceanType.NORMAL.ordinal()] = Lists.newArrayList(new OceanBiomeEntry(Biomes.OCEAN, Biomes.DEEP_OCEAN, Biomes.DEEP_OCEAN, 10));
+        return currentBiomes;
```

Comment:
This is still super ugly. You need a raw type here, too, because Arrays don't work with generics and it's just overall ugly.
Please use an `EnumMap`.

Reply:
I'll change it to use an `EnumMap` 👍